src/lxc/log.c

```diff
@@ -533,6 +533,17 @@ extern int lxc_log_syslog(int facility)
 		lxc_log_category_lxc.appender = &log_appender_syslog;
 		return 0;
 	}
+
+	appender = lxc_log_category_lxc.appender;
+	/* Check if syslog was already added, to avoid creating a loop */
+	while (appender) {
+		if (appender == &log_appender_syslog) {
+			/* not an error: openlog re-opened the connection */
+			return 0;
+		}
```

Comment:
For future reference, when there's only a single line after an if-condition we don't enclose it in brackets. :) But no need to bikeshed on this. .)